packages/@sanity/form-builder/src/inputs/BooleanInput.tsx

```diff
@@ -24,43 +23,43 @@ const BooleanInput = React.forwardRef(
       [onChange]
     )
 
+    const indeterminate = typeof value !== 'boolean'
+    const checked = value || false
+
+    const Input = layout === 'checkbox' ? Checkbox : Switch
```

Comment:
Super minor this but maybe we should call this something that signifies more clearly that it can vary between different components? 🙂 Like `CheckboxOrSwitch` or `LayoutSpecificInput` or something like that just to increase readability? Could perhaps even wrap it into a separate mini-component to make it more clearly that it branches into two paths.